api/src/test/java/org/openmrs/module/commonreports/reports/VaccinationReportManagerTest.java

```diff
@@ -0,0 +1,138 @@
+package org.openmrs.module.commonreports.reports;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.io.File;
+import java.sql.SQLException;
+import java.util.Iterator;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.openmrs.Cohort;
+import org.openmrs.module.commonreports.reports.BaseModuleContextSensitiveMysqlBackedTest;
+import org.openmrs.module.commonreports.ActivatedReportManager;
+import org.openmrs.module.commonreports.CommonReportsConstants;
+import org.openmrs.module.initializer.Domain;
+import org.openmrs.module.initializer.api.InitializerService;
+import org.openmrs.module.initializer.api.loaders.Loader;
+import org.openmrs.module.reporting.common.DateUtil;
+import org.openmrs.module.reporting.dataset.DataSetRow;
+import org.openmrs.module.reporting.evaluation.EvaluationContext;
+import org.openmrs.module.reporting.report.ReportData;
+import org.openmrs.module.reporting.report.definition.ReportDefinition;
+import org.openmrs.module.reporting.report.definition.service.ReportDefinitionService;
+import org.openmrs.module.reporting.report.manager.ReportManagerUtil;
+import org.openmrs.module.reporting.report.service.ReportService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+public class VaccinationReportManagerTest extends BaseModuleContextSensitiveMysqlBackedTest {
+	
+	public VaccinationReportManagerTest() throws SQLException {
+		super();
+		// TODO Auto-generated constructor stub
+	}
+	
+	@Autowired
+	private InitializerService iniz;
+	
+	@Autowired
+	private ReportService rs;
+	
+	@Autowired
+	private ReportDefinitionService rds;
+	
+	@Autowired
+	@Qualifier(CommonReportsConstants.COMPONENT_REPORTMANAGER_VACCINATION)
+	private ActivatedReportManager manager;
+	
+	@Before
+	public void setUp() throws Exception {
+		executeDataSet("org/openmrs/module/reporting/include/ReportTestDataset-openmrs-2.0.xml");
+		executeDataSet("org/openmrs/module/commonreports/include/vaccinationTestDataset.xml");
+		
+		String path = getClass().getClassLoader().getResource("testAppDataDir").getPath() + File.separator;
+		System.setProperty("OPENMRS_APPLICATION_DATA_DIRECTORY", path);
+		
+		for (Loader loader : iniz.getLoaders()) {
+			if (loader.getDomainName().equals(Domain.JSON_KEY_VALUES.getName())) {
+				loader.load();
+			}
+		}
+	}
+	
+	@Test
+	public void setupReport_shouldSetUpReport() {
+		
+		// replay
+		ReportManagerUtil.setupReport(manager);
+		
+		// verify
+		Assert.assertNotNull(rs.getReportDesignByUuid("ab8b5d26-8d13-4397-9690-0f107ad50adf"));
+		
+	}
+	
+	@Test
+	public void testReport() throws Exception {
+		
+		EvaluationContext context = new EvaluationContext();
+		context.addParameterValue("startDate", DateUtil.parseDate("2021-05-01", "yyyy-MM-dd"));
+		context.addParameterValue("endDate", DateUtil.parseDate("2021-07-30", "yyyy-MM-dd"));
```

Comment:
You want to check whether the dataset dates match with the reporting window defined here. For example compare with [this](https://github.com/mekomsolutions/openmrs-module-commonreports/pull/37/files#diff-f6ed40f3fd61c72dfc2269b8116e0cce656e5de17462bce9e414bbded46ca646R56-R57).